File: src/main/java/no/novari/flyt/mapping/kafka/configuration/KafkaEventProperties.java
package no.novari.flyt.mapping.kafka.configuration;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.time.Duration;

@Getter
@Setter
@Component
@ConfigurationProperties(prefix = "novari.flyt.mapping-service.kafka.event")
public class KafkaEventProperties {

    private Duration instanceProcessingEventsRetentionTime;
}

File: src/main/kotlin/no/novari/flyt/mapping/kafka/configuration/KafkaEventProperties.kt
package no.novari.flyt.mapping.kafka.configuration

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import java.time.Duration

@Component
@ConfigurationProperties(prefix = "novari.flyt.mapping-service.kafka.event")
class KafkaEventProperties {
    var instanceProcessingEventsRetentionTime: Duration = Duration.ofDays(4)
}
